tests/data/audience_spec.json

```diff
@@ -0,0 +1,101 @@
+[
+  {
+    "suite_name": "string primitives",
+    "test_cases": [
+      { "audience_json": ["equals", "", ""], "expect_result": true },
+      { "audience_json": ["equals", "foo", ""], "expect_result": false },
+      { "audience_json": ["equals", "", "bar"], "expect_result": false },
+      { "audience_json": ["equals", "foo", "foo"], "expect_result": true },
+      { "audience_json": ["equals", "foo", "bar"], "expect_result": false },
+      { "audience_json": ["contains", "foo", ""], "expect_result": true },
+      { "audience_json": ["contains", "", "foo"], "expect_result": false },
+      { "audience_json": ["contains", "foo", "o"], "expect_result": true },
+      { "audience_json": ["contains", "foo", "a"], "expect_result": false },
+      { "audience_json": ["matches", "foo", ""], "expect_result": true },
+      { "audience_json": ["matches", "", "foo"], "expect_result": false },
+      { "audience_json": ["matches", "foo", "(bar|foo)"], "expect_result": true },
+      { "audience_json": ["matches", "foo", "aa*"], "expect_result": false },
```

Comment:
As I was reviewing the primitive implementations, I had to refer back here to correctly identify the first and second argument meaning. Perhaps we could add a couple more test cases as well:

```
{ "audience_json": ["matches", "", "aa*"], "expect_result": false },
{ "audience_json": ["matches", "a", "aa*"], "expect_result": true },
{ "audience_json": ["matches", "aa", "aa*"], "expect_result": true },
{ "audience_json": ["matches", "aaa", "aa*"], "expect_result": true },
```

What do you think?

Reply:
Good point, adding it to the test cases!